### PR TITLE
RDM-4781-Data not being shown on multi-select list

### PR DESCRIPTION
**Pull request by ganganna-peddapothu**
### Data not being shown on multi-select list using AND and CONTAINS logic ###

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X ] No
```

### CODE REVIEW

**Comment by ganganna-peddapothu** [comment]
> this solution could work. But I think it relies on the fact that we always navigate to the previous page. Which is something we might change in the future if we add the ability to navigate forward to selected pages.
> I think the issue here is that we are passing to the page show condition evaluation only the form fields.
> We should instead pass both the form fields and the event fields. This way the show condition has available also the initial values of the fields. Which now is missing and therefore causing the show condition not to be properly evaluated

**Comment by ganganna-peddapothu** [comment]
Addressed

File: src/shared/services/fields/fields.purger.ts [hunk 1]
```diff
@@ -94,4 +94,8 @@ export class FieldsPurger {
   private isReadonly(case_field: CaseField) {
     return case_field.display_context.toUpperCase() === 'READONLY'
   }
+
+  private findIndex(wizardPageId, wizard) {
```
Comment (mario-paniccia) [inline src/shared/services/fields/fields.purger.ts]: leftover?

File: src/shared/services/fields/fields.utils.ts [hunk 2]
```diff
@@ -127,6 +127,9 @@ export class FieldsUtils {
       return page.parsedShowCondition.match(currentState);
     };
   }
+  public getCurrentEventState(eventTrigger, form): any {
```
Comment (mario-paniccia) [inline src/shared/services/fields/fields.utils.ts]: can you change the methods in this class now to call getCurrentEventState? So we remove duplication

File: src/shared/services/fields/fields.purger.ts [hunk 3]
```diff
@@ -37,23 +37,23 @@ export class FieldsPurger {
...
+        if (this.isHidden(condition, formFields.data) && !(this.isReadonly(case_field))) {
           this.resetField(form, case_field);
         }
       }
     });
   }
 
   private isHidden(condition, formFields) {
-    return !condition.match(formFields.data);
+    return !condition.match(formFields);
```
Comment (mario-paniccia) [inline src/shared/services/fields/fields.purger.ts]: what's the reason of this change?

File: src/shared/services/fields/fields.purger.ts [hunk 4]
```diff
@@ -37,23 +37,23 @@ export class FieldsPurger {
...
   }
 
   private findCaseFieldByWizardPageFieldId(currentPage, wizardPageField) {
     return currentPage.case_fields.find(cf => cf.id === wizardPageField.case_field_id);
   }
 
   private hasShowConditionPage(wizardPage, formFields): boolean {
-    return wizardPage.show_condition && formFields.data[this.getShowConditionKey(wizardPage.show_condition)];
+    return wizardPage.show_condition && formFields[this.getShowConditionKey(wizardPage.show_condition)];
```
Comment (mario-paniccia) [inline src/shared/services/fields/fields.purger.ts]: what's the reason of this change?

Comment (ganganna-peddapothu) [reply src/shared/services/fields/fields.purger.ts]: We are directly getting data fields. We don't have to call again data property.

Comment (mario-paniccia) [reply src/shared/services/fields/fields.purger.ts]: ok got it cheers :)
